### PR TITLE
Intgr 215

### PR DESCRIPTION
**Pull request by kmets89**

### CODE REVIEW

File: src/pages/myGeotab/apiReference/object.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import "./reference.scss";
+
+export default function Object() {
+    const { objectId = "" } = useParams();
+    const storedObjectData = JSON.parse(sessionStorage[objectId]);
+    const objectDescription = storedObjectData.description;
+    const properties = storedObjectData.properties;
+
+    const PropertyTable: ReactNode = <div className="paragraph">
+        <table className="table-container">
```
Comment (kmets89) [inline src/pages/myGeotab/apiReference/object.tsx:18]: this isn't in a table in the figma, did something change?

File: src/pages/myGeotab/apiReference/method.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    const TryMeCodeBlock: ReactNode = <div className="paragraph">
+        <CodeSample
+        language="javascript"
+        code={codeSample}></CodeSample>
+    </div>
+
+    const pageTitle: PageTitleProps = {
+        "title": methodId,
+        "breadCrumbItems": ["MYG", "API Reference", "Methods", methodId]
```
Comment (kmets89) [inline src/pages/myGeotab/apiReference/method.tsx:44]: figma shows (...) after each method name, in breadcrumb and title

File: src/pages/myGeotab/apiReference/methods.tsx [hunk 3]
```diff
@@ -1,7 +1,59 @@
...
+        "details": RenderStringWithUrl(methodDetails[1].description)
+    };
+
+    pageSections.push(pageSectionObject);
+
     return (
-        <div>
-            <h1>Just so many methods</h1>
+        <div id={methodDetails[0]}>
+            <Header title={methodDetails[0]}>
```
Comment (kmets89) [inline src/pages/myGeotab/apiReference/methods.tsx:38]: missing (...) after title

File: src/pages/myGeotab/apiReference/myGParser.ts
Comment (kmets89) [inline src/pages/myGeotab/apiReference/myGParser.ts:1]: async needs to be removed from method names eg: AuthenticateAsync should be Authenticate
entries should be ordered alphabetically
API object is missing from objects page

File: package-lock.json
Comment (jyalenkatian) [inline package-lock.json:1]: This file needs to be in the gitignore

File: .npmrc
Comment (jyalenkatian) [inline .npmrc:1]: This files needs to be in the gitignore

File: src/pages/myGeotab/apiReference/method.tsx [hunk 4]
```diff
@@ -1,12 +1,84 @@
...
+
+    const TryMeCodeBlock: ReactNode = <div className="paragraph">
+        <CodeSample
+        language="javascript"
+        code={codeSample}></CodeSample>
+    </div>
+
+    const pageTitle: PageTitleProps = {
+        "title": methodId + "(...)",
+        "breadCrumbItems": ["MYG", "API Reference", "Methods", methodId + "(...)"]
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/method.tsx:44]: There should be a space before the (...)

File: src/pages/myGeotab/apiReference/method.tsx [hunk 5]
```diff
@@ -1,12 +1,84 @@
...
 
 export default function Method() {
-    const { methodId } = useParams();
+    const { methodId = "" } = useParams();
+    const storedMethodData = JSON.parse(sessionStorage[methodId]);
+    const parameters = storedMethodData.param;
+    const returnDescription = storedMethodData.returns;
+    const codeSample = storedMethodData.example;
+    
+    const ParamTable: ReactNode = <div className="paragraph">
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/method.tsx:19]: This variable should be camelCase & the content should be inside parentheses.

File: src/pages/myGeotab/apiReference/method.tsx [hunk 6]
```diff
@@ -1,12 +1,84 @@
...
+                        <tr>
+                            <td>{parameter.name}</td>
+                            <td>{RenderStringWithUrl(parameter.description)}</td>
+                        </tr>
+                    )}
+            </tbody>
+        </table>
+    </div>
+
+    const ReturnDescription: ReactNode = <div className="paragraph">
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/method.tsx:32]: This variable should be camelCase & the content should be inside parentheses.

File: src/pages/myGeotab/apiReference/method.tsx [hunk 7]
```diff
@@ -1,12 +1,84 @@
...
+        <p>{RenderStringWithUrl(returnDescription)}</p>
+    </div>
+
+    const TryMeCodeBlock: ReactNode = <div className="paragraph">
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/method.tsx:36]: This variable should be camelCase & the content should be inside parentheses.

File: src/pages/myGeotab/apiReference/methods.tsx [hunk 8]
```diff
@@ -36,8 +37,8 @@ const methodItems = methods.map((methodDetails: any) => {
     return (
         <div className="paragraph" id={methodDetails[0]}>
             <h3 className="methods__method-title">
-                {methodDetails[0]}
-                <Link to={`/method/${methodDetails[0]}`} className="methods__view-button">
+                {methodDetails[0] + "(...)"}
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methods.tsx:40]: There should be a space before (...)

File: src/pages/myGeotab/apiReference/object.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    const propertyParagaphs: ReactNode = (
+        <div className="paragraph">
+            {properties.map((property: any) =>
+                <div>
+                    <h3>{property.name}</h3>
+                    <p>{RenderStringWithUrl(property.description)}</p>
+                </div>
+            )}
+        </div>
+    )
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/object.tsx:26]: Missing semicolon

File: src/pages/myGeotab/apiReference/object.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+    const pageTitle: PageTitleProps = {
+        "title": objectId,
+        "breadCrumbItems": ["MYG", "API Reference", "Objects", objectId]
+    };
+
+    const pageSections: TableOfContentsItem[] = [
+        {
+            "elementId": "introduction",
+            "summary": "introduction"
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/object.tsx:36]:
```suggestion
            "summary": "Introduction"
```

File: src/pages/myGeotab/apiReference/methods.tsx [hunk 11]
```diff
@@ -14,15 +15,15 @@ request.send();
 let xml: any = request.responseXML;
 
 const pageTitle: PageTitleProps = {
-    "title": "Methods",
-    "breadCrumbItems": ["MYG", "API Reference", "Methods"]
+    "title": "METHODS",
+    "breadCrumbItems": ["MYG", "API REFERENCE", "METHODS"]
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methods.tsx:19]: Revert casing. Should be how it was before.

File: src/pages/myGeotab/apiReference/methods.tsx
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/methods.tsx:1]: Line 50 should have JSX.Element return type for component

File: src/pages/myGeotab/apiReference/object.tsx [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+import { PageTitleProps } from "../../../components/PageTitle/PageTitle";
+import { HeaderSections } from "../../../components/Header/headerSectionsEnum";
+import { TableOfContentsItem } from "../../../components/TableOfContents/TableOfContents";
+import "./reference.scss";
+
+export default function Object() {
+    const { objectId = "" } = useParams();
+    const storedObjectData = JSON.parse(sessionStorage[objectId]);
+    const objectDescription = storedObjectData.description;
+    const properties = storedObjectData.properties;
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/object.tsx:14]: Add data types for variables.

File: src/pages/myGeotab/apiReference/object.tsx [hunk 13]
```diff
@@ -0,0 +1,56 @@
+import { ReactNode } from "react";
+import RenderStringWithUrl from './renderStringWithUrl';
+import { useParams } from "react-router-dom";
+import { Page } from "../../../components";
+import { PageTitleProps } from "../../../components/PageTitle/PageTitle";
+import { HeaderSections } from "../../../components/Header/headerSectionsEnum";
+import { TableOfContentsItem } from "../../../components/TableOfContents/TableOfContents";
+import "./reference.scss";
+
+export default function Object() {
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/object.tsx:10]:
```suggestion
export default function Object():JSX.Element {
```

File: src/pages/myGeotab/apiReference/object.tsx [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+                    <h3>{property.name}</h3>
+                    <p>{RenderStringWithUrl(property.description)}</p>
+                </div>
+            )}
+        </div>
+    );
+
+    const pageTitle: PageTitleProps = {
+        "title": objectId.toUpperCase(),
+        "breadCrumbItems": ["MYG", "API REFERENCE", "OBJECTS", objectId.toUpperCase()]
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/object.tsx:35]: Shouldn't be all uppercase, title casing instead.

File: src/pages/myGeotab/apiReference/object.tsx [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+            "elementId": "properties",
+            "summary": "Properties",
+            "details": propertyParagaphs
+        }
+    ];
+
+    return (
+        <Page section={HeaderSections.MyGeotab} pageTitle={pageTitle} tableOfContents={pageSections}>
+
+        </Page>
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/object.tsx:54]: Can be self closing
```suggestion
        <Page section={HeaderSections.MyGeotab} pageTitle={pageTitle} tableOfContents={pageSections} />
```

File: src/pages/myGeotab/apiReference/objects.tsx [hunk 16]
```diff
@@ -14,15 +15,15 @@ request.send();
 let xml: any = request.responseXML;
 
 const pageTitle: PageTitleProps = {
-    "title": "Objects",
-    "breadCrumbItems": ["MYG", "API Reference", "Objects"]
+    "title": "OBJECTS",
+    "breadCrumbItems": ["MYG", "API RERFENCE", "OBJECTS"]
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/objects.tsx:19]: Revert

File: src/pages/myGeotab/apiReference/objects.tsx [hunk 17]
```diff
@@ -46,7 +47,6 @@ const objectItems = objects.map((objectDetails: any) => {
     )
 })
 
-
 export default function Objects() {
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/objects.tsx:50]:
```suggestion
export default function Objects(): JSX.Element {
```

File: src/pages/myGeotab/apiReference/objects.tsx [hunk 18]
```diff
@@ -14,15 +15,15 @@ request.send();
...
 
 const pageSections: TableOfContentsItem[] = [
 
 ];
 
-const objects = Object.entries(myGParser(xml, 'object', ['T:Geotab.Checkmate.ObjectModel', 'P:Geotab.Checkmate.ObjectModel']));
+const objects = Object.entries(myGParser(xml, 'object', ['T:Geotab.Checkmate.ObjectModel', 'T:Geotab.Checkmate.API','P:Geotab.Checkmate.ObjectModel', 'M:Geotab.Checkmate.API.#ctor', 'F:Geotab.Checkmate.ObjectModel'])).sort(sortAlphabetical);
 const objectItems = objects.map((objectDetails: any) => {
     sessionStorage.setItem(objectDetails[0], JSON.stringify(objectDetails[1]));
     let pageSectionObject = {
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/objects.tsx:29]:
```suggestion
    let pageSectionObject: TableOfContentsItem = {
```

File: src/pages/myGeotab/apiReference/objects.tsx [hunk 19]
```diff
@@ -14,15 +15,15 @@ request.send();
...
+    "breadCrumbItems": ["MYG", "API RERFENCE", "OBJECTS"]
 };
 
 const pageSections: TableOfContentsItem[] = [
 
 ];
 
-const objects = Object.entries(myGParser(xml, 'object', ['T:Geotab.Checkmate.ObjectModel', 'P:Geotab.Checkmate.ObjectModel']));
+const objects = Object.entries(myGParser(xml, 'object', ['T:Geotab.Checkmate.ObjectModel', 'T:Geotab.Checkmate.API','P:Geotab.Checkmate.ObjectModel', 'M:Geotab.Checkmate.API.#ctor', 'F:Geotab.Checkmate.ObjectModel'])).sort(sortAlphabetical);
 const objectItems = objects.map((objectDetails: any) => {
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/objects.tsx:27]: `any` should be replaced with the actual type here. objects & objectItems should also have a type specified.

File: src/pages/myGeotab/apiReference/renderStringWithUrl.tsx [hunk 20]
```diff
@@ -1,3 +1,5 @@
+import { HashLink } from 'react-router-hash-link';
+
 export default function RenderStringWithLinks(text: string) {
```
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/renderStringWithUrl.tsx:3]: Specify what data type is being returned from this function and the same for the rest of the variables/loops/functions in this file.

File: src/pages/myGeotab/apiReference/myGParser.ts
Comment (jyalenkatian) [inline src/pages/myGeotab/apiReference/myGParser.ts:1]: Types for variables throughout the file need to be specified.

**Review by jyalenkatian — CHANGES_REQUESTED** [review]
@wajihqazi94 Comments have been left in the PR.

**Review by jyalenkatian — APPROVED** [review]

**Pull request merged**